### PR TITLE
New package: snort-2.9.8.2, daq-2.0.6, libdnet-1.11

### PR DESCRIPTION
**Pull request by subsonik**
Snort and its two dependencies. Will need to add pulledpork later as a new package.

### CODE REVIEW

**Comment by subsonik** [comment]
Need to integrate patches to fix musl build, I'll use the ones from Alpine Linux.

**Comment by Vaelatern** [comment]
The rule within Void is that for every template altered, there should be one commit. Those commits should be named as directed by the end of the [Manual (link)](https://github.com/voidlinux/void-packages/blob/master/Manual.md#contributing-via-git).

Only one PR, please, unless you are told otherwise. Unrelated features, such as if you fixed something in `firefox` and added a new spotify client, should go in separate PRs. If you wanted to package something, and it depended on a different package not yet in Void, you would put both of those package additions in the **same** PR.

**Comment by subsonik** [comment]
Thanks for your help, @Vaelatern. I'm still getting acquainted with this whole process, it doesn't help that I'm also new to using GitHub as a committer. It's a bit overwhelming but I'm getting there. :)

**Comment by Vaelatern** [comment]
`nocross=yes`

File: srcpkgs/daq/template [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+distfiles="${homepage}/downloads/snort/${pkgname}-${version}.tar.gz"
+checksum=b40e1d1273e08aaeaa86e69d4f28d535b7e53bdb3898adf539266b63137be7cb
+disable_parallel_build=yes
+nocross="./configure fails when cross-compiling; cannot run test program"
+
+pre_configure() {
+	case "$XBPS_TARGET_MACHINE" in
+		*-musl) patch -Np1 -i ${FILESDIR}/fix-musl-build.patch;;
+	esac
+}
```
Comment (xtraeme) [inline srcpkgs/daq/template]: This is nonsense... move the patches to the "patches" directory and remove this function completely.

File: srcpkgs/daq/template [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+hostmakedepends="bison flex"
+makedepends="libpcap-devel"
+short_desc="Snort's abstraction layer for libpcap"
+maintainer="John Rowley <enterthevoid@codesector.co>"
+license="GPL-2"
+homepage="https://www.snort.org"
+distfiles="${homepage}/downloads/snort/${pkgname}-${version}.tar.gz"
+checksum=b40e1d1273e08aaeaa86e69d4f28d535b7e53bdb3898adf539266b63137be7cb
+disable_parallel_build=yes
+nocross="./configure fails when cross-compiling; cannot run test program"
```
Comment (xtraeme) [inline srcpkgs/daq/template]: Please stop abusing "nocross", if the pkg fails to cross compile just leave it as is.

File: srcpkgs/daq/template [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Template file for 'daq'
+pkgname=daq
+version=2.0.6
+revision=1
+build_style=gnu-configure
+hostmakedepends="bison flex"
```
Comment (xtraeme) [inline srcpkgs/daq/template]: flex depends on bison, bison is redundant.

File: srcpkgs/libdnet/template [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'libdnet'
+pkgname=libdnet
+version=1.11
+revision=1
+build_style=gnu-configure
+hostmakedepends="pkg-config"
+short_desc="A portable interface to several low-level networking routines"
+maintainer="John Rowley <enterthevoid@codesector.co>"
+license="custom"
```
Comment (xtraeme) [inline srcpkgs/libdnet/template]: "custom" is not a proper license, please set it specifically.

File: srcpkgs/snort/template [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+hostmakedepends="flex"
+makedepends="pcre-devel libpcap-devel libdnet-devel daq-devel libtirpc-devel"
+short_desc="A lightweight network intrusion detection system"
+maintainer="John Rowley <enterthevoid@codesector.co>"
+license="GPL-2"
+homepage="https://www.snort.org"
+distfiles="${homepage}/downloads/snort/${pkgname}-${version}.tar.gz"
+checksum=4075012d350dfa47a0200b7a920323f15cb7c370790f2a47367c03aba4009333
+conf_files="/etc/snort/*"
+system_groups="snort:29"
```
Comment (xtraeme) [inline srcpkgs/snort/template]: `system_groups` is redundant here, because `system_accounts` creates a system user/group by default.

**Comment by radare** [comment]
oh right. lgtm

**Comment by Vaelatern** [comment]
Please rebase on master to remove the conflicts and trigger a travis rebuild.

**Comment by ebfe** [comment]
Needs another rebase unfortunately.

**Pull request closed without merging**